### PR TITLE
ServiceAccount without ImagePullSecrets should not fail to fetch

### LINKED ISSUE #584: Entrypoint remote image fetch fails for a non-default service account with no `ImagePullSecrets`
# Expected Behavior

Publicly accessible images should be usable when using a non-default service account, even if it has no `ImagePullSecret`s defined.

# Actual Behavior

All attempts to fetch remote images to handle their entrypoints with a non-default service account without any `ImagePullSecret`s fail like:
```
Failed to fetch remote image SOME_IMAGE:
No ImagePullSecret for service account SA_NAME: <nil>'
```

# Steps to Reproduce the Problem

1. Create a `Pipeline` with a `Task` using a publicly accessible image.
2. Create a `PipelineRun` with an explicit service account, without adding any `ImagePullSecret`s to the service account, using that `Pipeline`.
3. Check the `PipelineRun`'s status.

# Additional Info

### PR DESCRIPTION
**Pull request by abayer**
# Changes

While `ImagePullSecret`s would still be needed for any private
registry, not having any on the specified service account shouldn't
cause a failure to even try to fetch the image.

Fixes #584

# Submitter Checklist

These are the criteria that every PR should meet, please check them off as you
review them:

- [x] Includes [tests](https://github.com/knative/build-pipeline/blob/master/CONTRIBUTING.md#principles) (if functionality changed/added)
- [x] Includes [docs](https://github.com/knative/build-pipeline/blob/master/CONTRIBUTING.md#principles) (if user facing)
- [x] Commit messages follow [commit message best practices](https://github.com/knative/build-pipeline/blob/master/CONTRIBUTING.md#commit-messages)

# Release Notes

n/a

### CODE REVIEW

**Comment by pivotal-nader-ziada** [comment]
actually @abayer I think if you remove all that code and just have `k8schain.New` while passing the `serviceaccount` the library will do everything for you.
Check this here https://github.com/google/go-containerregistry/blob/master/pkg/authn/k8schain/k8schain.go#L55

what do you think?

**Comment by abayer** [comment]
@pivotal-nader-ziada Nice! Switching now.

**Comment by abayer** [comment]
ha, this will also deal with cases of multiple `ImagePullSecret`s on a non-default service account, which currently would result in just the last one encountered being used.

File: pkg/reconciler/v1alpha1/taskrun/entrypoint/entrypoint.go [hunk 1]
```diff
@@ -187,38 +186,18 @@ func getRemoteImage(image string, kubeclient kubernetes.Interface, build *buildv
...
-		}
-	} else {
-		kc, err = k8schain.New(kubeclient, k8schain.Options{
-			Namespace:          build.Namespace,
-			ServiceAccountName: "default",
-		})
-		if err != nil {
-			return nil, fmt.Errorf("Failed to create k8schain: %v", err)
-		}
+	if serviceAccountName == "" {
```
Comment (shashwathi) [inline pkg/reconciler/v1alpha1/taskrun/entrypoint/entrypoint.go]: Library does this piece of code as well. Pass the `serviceAccountName ` (empty or not) to k8schain library

Comment (abayer) [reply pkg/reconciler/v1alpha1/taskrun/entrypoint/entrypoint.go]: Whoops, I actually had that thought and then it slipped my mind. Updated.

**Comment by pivotal-nader-ziada** [comment]
Thanks @abayer and @shashwathi for the quick fix

**Comment by abayer** [comment]
@pivotal-nader-ziada Happy to help! I'm glad I started work on switching Jenkins X over as early as I did, or who knows when we would have found this.

**Pull request merged**